src/eval_framework/llm/base.py

```diff
@@ -56,3 +61,55 @@ def generate(
     ) -> list[RawCompletion]:
         messages: list[Sequence[Message]] = [sample.messages for sample in samples]
         return self.generate_from_messages(messages, stop_sequences, max_tokens, temperature)
+
+    @contextmanager
```

Comment:
Suggested alternative: This is just a normal function which accepts wandb_fs as a parameter. Then you also don't need any of this try-except and `WandbFs.cleanup()` in general. Then it's used as:
```
with WandbFs(user_supplied_download_path=user_supplied_download_path) as wandb_fs:
        self.LLM_NAME = self.download_wandb_artifact(artifact_name, version, wandb_fs) as:
        ...
```